File: musifanAngular/src/app/models/utilisateur.ts
import { Artiste } from './artiste';
import { Concert } from './concert';
import { Byte } from '@angular/compiler/src/util';

export class Utilisateur {
  public constructor(
    private _id?: number | undefined,
    private _mail?: string | undefined,
    private _nom?: string | undefined,
    private _prenom?: string | undefined,
    private _pseudo?: string | undefined,
    private _telephone?: string | undefined,
    private _artistes?: Artiste[] | undefined,
    private _concerts?: Concert[] | undefined,
    private _photoProfil?: Byte[] | undefined
  ) {}

  /**
   * Getter id
   * @return {number | undefined}
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter mail
   * @return {string | undefined}
   */
  public get mail(): string | undefined {
    return this._mail;
  }

  /**
   * Getter nom
   * @return {string | undefined}
   */
  public get nom(): string | undefined {
    return this._nom;
  }

  /**
   * Getter prenom
   * @return {string | undefined }
   */
  public get prenom(): string | undefined {
    return this._prenom;
  }

  /**
   * Getter pseudo
   * @return {string | undefined}
   */
  public get pseudo(): string | undefined {
    return this._pseudo;
  }

  /**
   * Getter telephone
   * @return {string | undefined}
   */
  public get telephone(): string | undefined {
    return this._telephone;
  }

  /**
   * Getter photoProfil
   * @return {Byte[] | undefined}
   */
  public get photoProfil(): Byte[] | undefined {
    return this._photoProfil;
  }

  /**
   * Getter artistes
   * @return {Artiste[] | undefined }
   */
  public get artistes(): Artiste[] | undefined {
    return this._artistes;
  }

  /**
   * Getter concerts
   * @return {Concert[] | undefined }
   */
  public get concerts(): Concert[] | undefined {
    return this._concerts;
  }

  /**
   * Setter id
   * @param {number | undefined} value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter mail
   * @param {string | undefined} value
   */
  public set mail(value: string | undefined) {
    this._mail = value;
  }

  /**
   * Setter nom
   * @param {string | undefined} value
   */
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  /**
   * Setter prenom
   * @param {string | undefined} value
   */
  public set prenom(value: string | undefined) {
    this._prenom = value;
  }

  /**
   * Setter pseudo
   * @param {string | undefined} value
   */
  public set pseudo(value: string | undefined) {
    this._pseudo = value;
  }

  /**
   * Setter telephone
   * @param {string | undefined} value
   */
  public set telephone(value: string | undefined) {
    this._telephone = value;
  }

  /**
   * Setter photoProfil
   * @param {Byte[] | undefined} value
   */
  public set photoProfil(value: Byte[] | undefined) {
    this._photoProfil = value;
  }

  /**
   * Setter artistes
   * @param {Artiste[] | undefined} value
   */
  public set artistes(value: Artiste[] | undefined) {
    this._artistes = value;
  }

  /**
   * Setter concerts
   * @param {Concert[] | undefined} value
   */
  public set concerts(value: Concert[] | undefined) {
    this._concerts = value;
  }
}

File: musifan/src/main/java/musifan/musifan/dto/Utilisateur.java
package musifan.musifan.dto;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.OneToMany;

import com.fasterxml.jackson.annotation.JsonView;

import musifan.musifan.dto.Commande;
import musifan.musifan.entity.JsonViews;
import musifan.musifan.entity.LigneUtilisateur;

public class Utilisateur {
	@JsonView({JsonViews.Common.class,})
	private Long id;
	@JsonView({JsonViews.Common.class,})
	private String mail;
	private String password;
	@JsonView({JsonViews.Common.class,})
	private String nom;
	@JsonView({JsonViews.Common.class,})
	private String prenom;
	@JsonView({JsonViews.Common.class,})
	private String telephone;
	@JsonView({JsonViews.Common.class,})
	private Byte[] photoProfil;
	@JsonView({JsonViews.Common.class,})
	private String pseudo;
	@JsonView({JsonViews.UtilisateurAvecCommandes.class,})
	private Set<Commande> commande = new HashSet<Commande>();
	@JsonView({JsonViews.UtilisateurAvecArtiste.class,})
	private Set<Artiste> artistes = new HashSet<Artiste>();
	
	public Utilisateur() {
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getMail() {
		return mail;
	}

	public void setMail(String mail) {
		this.mail = mail;
	}

	public String getPassword() {
		return password;
	}

	public void setPassword(String password) {
		this.password = password;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getPrenom() {
		return prenom;
	}

	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	public String getTelephone() {
		return telephone;
	}

	public void setTelephone(String telephone) {
		this.telephone = telephone;
	}

	public Byte[] getPhotoProfil() {
		return photoProfil;
	}

	public void setPhotoProfil(Byte[] photoProfil) {
		this.photoProfil = photoProfil;
	}

	public String getPseudo() {
		return pseudo;
	}

	public void setPseudo(String pseudo) {
		this.pseudo = pseudo;
	}

	public Set<Commande> getCommande() {
		return commande;
	}

	public void setCommande(Set<Commande> commande) {
		this.commande = commande;
	}

	public Set<Artiste> getArtistes() {
		return artistes;
	}

	public void setArtistes(Set<Artiste> artistes) {
		this.artistes = artistes;
	}
	
	

}
